docs/design/proposals/rgd-ownership-and-deletion.md

```diff
@@ -0,0 +1,203 @@
+
+```yaml
+metadata:
+  name: sample.kro.run
+  ownerReferences:
+    - apiVersion: kro.run/v1alpha1
+      kind: ResourceGraphDefinition
+      name: sample
+      uid: 12345-abcde
+      controller: true
+      blockOwnerDeletion: true
+```
+
+**Resulting behavior:**
+
+* When an RGD is deleted, its owned CRD is automatically garbage-collected. This is proper behavior
+  as we can still delete RGDs with policy "orphan" to preserve CRDs.
+* Manual CRD deletion (while RGD still exists) will trigger re-reconciliation, as ownership implies controller responsibility.
+  (we can and should think about warning or failing user deletion requests on CRDs managed by KRO)
+* The original RGD label remains for traceability, but lifecycle is now managed by native Kubernetes GC.
+
+Integration tests confirm the presence and correctness of `ownerReferences` (including `controller` and `blockOwnerDeletion` flags).
+
+#### 2. Deletion protection for RGDs
+
+Because RGD deletion now cascades to delete its CRD, an **admission policy** prevents accidental data loss.
+
+When `config.allowCRDDeletion` is `false` (default), Helm installs a **ValidatingAdmissionPolicy** and **Binding**:
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: {{ include "kro.fullname" . }}-crd-protection-policy
+spec:
+  matchConstraints:
+    resourceRules:
+      - apiGroups: ["kro.run"]
+        apiVersions: ["v1alpha1"]
+        operations: ["DELETE"]
+        resources: ["resourcegraphdefinitions"]
+  validations:
+    - expression: |
+        has(oldObject.metadata.annotations) &&
+        oldObject.metadata.annotations['{{ .Values.validation.admission.policy.rgd.annotation.key }}'] == 'true'
+      reason: Invalid
+      message: |
+        Deletion denied. To proceed, set annotation '{{ .Values.validation.admission.policy.rgd.annotation.key }}: "true"'.
+        Removing an RGD also deletes its CustomResourceDefinition and may cause data loss.
+```
+
+Users must explicitly annotate an RGD to confirm deletion:
+
+```yaml
+metadata:
+  annotations:
+    kro.run/allow-delete: "true"
+```
+
+This ensures intentional deletions only.
+
+#### 3. Controller-side deletion control (`allowCRDDeletion`)
+
+Previously, the flag `allowCRDDeletion` determined whether KRO would delete CRDs it had created if the RGD was removed or changed.
+Now, with `OwnerReferences` managing lifecycle automatically, the flag’s meaning is refined:
+
+* When `allowCRDDeletion: false`:
+  KRO still sets owner references, but **does not itself delete CRDs**.
+  Deletion is handled by Kubernetes GC only when the RGD is removed.
+  The ValidatingAdmissionPolicy remains active to prevent accidental RGD deletion.
+
+* When `allowCRDDeletion: true`:
+  KRO may directly delete CRDs it manages (e.g., during reconciliation or cleanup).
+  The Helm chart skips installation of the ValidatingAdmissionPolicy, allowing full manual control.
```

Comment:
I'm unsure I make sense of the semantics.
It feels weird to me to have `allowCRDDeletion: false` and that CRD would be deleted anyhow (by k8s).

There's also a case that could lead to inconsistent behaviour, when the RGD `Kind`, `Group` or `APIVersion` changes. In this case, when `allowCRDDeletion: false` we wouldn't delete the dangling CRD but the CRD would be deleted in case of RGD deletion.

The proposed behaviour and lifecycle sounds more like `confirmCRDDeletion` or `requireManualJudgementForCRDDeletion`
This semantics would also bring consistency when `Kind`, `Group` or `APIVersion` changes.
Maybe the validation webhook needs to be on the CRD itself then?